File: samples/kmp/shared/src/commonMain/kotlin/jp/kamusoft/ksdialogs/samples/kmp/ModelDialogViewModel.kt
package jp.kamusoft.ksdialogs.samples.kmp

import jp.kamusoft.ksdialogs.kmp.DialogViewModel

/**
 * Model Dialog の ViewModel。
 *
 * 結果型は [Boolean] で、完了操作は `true` を報告する。
 * 共有コードで定義したこのクラスがそのまま各 OS のレジストリの登録キーになる。
 *
 * 実体はレジストリの ViewModel factory が引数なしで作り、表示するメッセージは
 * 型を渡す表示の configure が入れる。
 *
 * 結果の報告口は OS ごとの Native API から供給されるため (core/ADR-0018)、
 * 共有コードのこのクラスは報告口を持たない。中身の View が表示中の自分に紐付いた報告口を引く。
 */
class ModelDialogViewModel : DialogViewModel<Boolean> {
    /** ダイアログに表示するメッセージ。表示の直前に設定する。 */
    var message: String = ""
}

File: samples/maui/KsDialogs.Sample.Maui/ModelDialogViewModel.cs
namespace KsDialogs.Sample.Maui;

/// <summary>
/// Model Dialog の ViewModel。
/// </summary>
/// <remarks>
/// 結果型は <see cref="bool"/> で、完了操作は <see langword="true"/> を報告する。
/// 表示するメッセージは型指定 show の configure から設定される。
/// 結果は中身から渡される報告口ではなく、この ViewModel 自身が <c>Notifier</c> から報告する
/// (core/ADR-0018)。
/// </remarks>
public sealed class ModelDialogViewModel : IDialogViewModel
{
    /// <summary>ダイアログに表示するメッセージ。表示の直前に configure から設定する。</summary>
    public string Message { get; set; } = string.Empty;

    /// <summary>完了を報告する。</summary>
    public void Complete() => this.Notifier?.Complete(true);

    /// <summary>キャンセルを報告する。</summary>
    public void Cancel() => this.Notifier?.Cancel();
}
